src/components/pagination/pagination.ts

```diff
@@ -0,0 +1,301 @@
+    }
+  }
+
+  changePage(newPage: number) {
+    let model = JSON.parse(JSON.stringify(this.model));
+    model.currentPage = newPage;
+    this.service.change(this.name, model);
+  }
+}
+
+@Component({
+  selector: 'md-pagination-items-per-page',
+  template: `
+   <span *ngIf="itemsPerPageBefore" class="md-pagination-items-per-page-label md-pagination-items-per-page-before">{{itemsPerPageBefore}}</span>
+   <select [(ngModel)]="model.itemsPerPage" (ngModelChange)="changePaginationLength($event)" class="md-pagination-length-select">
+      <option *ngFor="let length of itemsPerPageOptions" [value]="length">
+        {{length}}
+      </option>
+    </select>
+    <span *ngIf="itemsPerPageAfter" class="md-pagination-items-per-page-label md-pagination-items-per-page-after">{{itemsPerPageAfter}}</span>
+  `,
+  host: {
+    '[class.md-pagination-items-per-page]': 'true',
+    '[hidden]': '!itemsPerPageOptions?.length'
+  }
+})
+export class MdPaginationItemsPerPage extends AbstractPaginationSubComponent {
+
+  @Input()
+  name: string = 'default';
+
+  @Input()
+  model: IPaginationModel = {
+    currentPage: 0,
+    itemsPerPage: 0,
+    totalItems: 0
+  };
+
+  @Input('items-per-page-before')
+  itemsPerPageBefore: string = 'Rows per page:';
+
+  @Input('items-per-page-after')
+  itemsPerPageAfter: string;
+
+  @Input('items-per-page-options')
+  itemsPerPageOptions: Array<number> = [];
+
+  constructor(protected service: PaginationService) {
+    super(service);
+  }
+
+  changePaginationLength(value) {
+    let model = JSON.parse(JSON.stringify(this.model));
+    model.currentPage = 1;
+    model.itemsPerPage = parseInt(value);
+    this.service.change(this.name, model);
+  }
+
+}
+
+export interface IPaginationChange {
+  name: string;
+  target: string;
+  pagination: IPaginationModel;
+}
+
+@Component({
+  selector: 'md-pagination',
+  template: `
+    <ng-content></ng-content>
+    <md-pagination-items-per-page 
+      *ngIf="defaultDisplay && itemsPerPage"
+      [name]="name" 
+      [model]="model" 
+      [items-per-page]="itemsPerPage"
+      [items-per-page-before]="itemsPerPageBefore"
+      [items-per-page-after]="itemsPerPageAfter"
+      [items-per-page-options]="itemsPerPageOptions"></md-pagination-items-per-page>
+    <md-pagination-range
+      *ngIf="defaultDisplay && range"
+      [name]="name" 
+      [model]="model" 
+      [rangeFormat]="rangeFormat"></md-pagination-range>
+    <md-pagination-controls
+      *ngIf="defaultDisplay && controls"
+      [name]="name" 
+      [model]="model"></md-pagination-controls>
+  `,
+  directives: [MdPaginationRange, MdPaginationControls, MdPaginationItemsPerPage],
+  host: {
+    '[class.md-pagination]': 'true'
+  }
+})
+export class MdPagination implements AfterContentInit, AfterViewInit {
+
+  @Input()
+  name: string = 'default';
+
+  @Input()
+  model: IPaginationModel = {
+    currentPage: 0,
+    itemsPerPage: 0,
+    totalItems: 0
+  };
+
+  @Input()
```

Comment:
Angular styleguide now calling for these to be on the same line in "Style 05-12"
e.g. `@Input() range: boolean = true;`